### DISCUSSION TITLE
Need to change Ports filed in Services to not required

### DISCUSSION THREAD

**Discussion by davolia** [netbox-community/netbox#7710] [Q&A]
Hi

Is anyone know how to change Port(s) field in Services to not required?

![image](https://user-images.githubusercontent.com/25403463/139695690-cd05be09-0de3-48ad-b6a5-0094dbeb221c.png)

**Comment by davolia** [comment]
Nobody can help me?

**Comment by candlerb** [comment]
A service in Netbox is, by definition, something that listens on a TCP or UDP port (like "SMTP server" = TCP port 25). Therefore making the port number optional makes no sense.

If you want to say "all ports" then I imagine you can put "0-65535" or "1-65535"

**Reply by davolia** [reply]
Thank you for answer.

For example I have PHP-FPM and I need to specify it. But PHP-FPM is not listen on any port. I don't want to specify the port. How can I make this filed to not required. I think it need to change from python scripts.

**Reply by candlerb** [reply]
Netbox "Services" are not meant to represent "software packages installed".

There are several options I can think of:

1. [Tags](https://netbox.readthedocs.io/en/stable/models/extras/tag/) (e.g. tag "package_php_fpm")
2. Create a separate boolean [custom field](https://netbox.readthedocs.io/en/stable/customization/custom-fields/) for each service (enabled on device and VM)
3. Create a single [multi-select](https://netbox.readthedocs.io/en/stable/customization/custom-fields/#custom-selection-fields) custom field listing all possible services (enabled on device and VM)
4. Write a [plugin](https://netbox.readthedocs.io/en/stable/plugins/) which represents "packages" linked to a device or VM

**Reply by davolia** [reply]
Ok, thank you very much.
